### PR TITLE
[Backport 17.06] Circle CI migration to 2.0

### PR DESCRIPTION
**Pull request by fcrisciani**

### CODE REVIEW

**Comment by selansen** [comment]
back port is not straight forward :( 
LGTM

**Comment by fcrisciani** [comment]
@thaJeztah any comment on this?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM golang:1.10.2 as dev
```
Comment (thaJeztah) [inline Dockerfile]: We should match the version that's used to build 17.06

**Comment by fcrisciani** [comment]
@selansen 17.06 build with go 1.8.7, was addressing @thaJeztah comment.
@thaJeztah if you can bless, I will merge

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+version: 2
+
+defaults: &defaults
+  working_directory: ~/go/src/github.com/docker/libnetwork
+  docker:
+    - image: 'circleci/golang:1.10'
```
Comment (thaJeztah) [inline .circleci/config.yml]: oh! does this one have to be updated as well?

Comment (fcrisciani) [reply .circleci/config.yml]: no this one is irrelevant for us, I will check if I can change it to something minimal, we build everything in the container and that's the version that counts

Comment (thaJeztah) [reply .circleci/config.yml]: Alright! I'd say; let's go then 💃

**Pull request merged**